conans/test/functional/profiles/profiles_test.py

```diff
@@ -39,3 +39,37 @@ def test_env_var_without_variable_substitution(self):
         self.assertIn("bar=other;new value", client.out)
         self.assertIn("zet=$KK", client.out)
         print(client.out)
+
+    def test_config_parser(self):
+        from configparser import ConfigParser, BasicInterpolation
+
+        class CustomInterpolation(BasicInterpolation):
+            """Interpolation which expands environment variables in values."""
+
+            def before_get(self, parser, section, option, value, defaults):
+                rawval = parser.get(section, option, raw=True)
+                val = rawval % parser.defaults()
+                return val
+
+        parser1 = ConfigParser(interpolation=CustomInterpolation())
+        profile1 = """
```

Comment:
please ``textwrap.dedent`` the files contents (one indent more than the variable declaration)